nixpkgs.pwsafe fails to build (but possibly merely a dependency problem)

###### Checklist

<!-- Note that these are hard requirements -->

<!--
You can use the "Go to file" functionality on github to find the package
Then you can go to the history for this package
Find the latest "package_name: old_version -> new_version" commit
The "new_version" is the the current version of the package
-->
- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
<!--
Type the name of your package and try to find an open pull request for the package
If you find an open pull request, you can review it!
There's a high chance that you'll have the new version right away while helping the community!
-->
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls): when I search the pulls for `pwsafe`, I see [pull 73145](https://github.com/NixOS/nixpkgs/pull/73145).  It's not obvious to me why this is relevant, but see below

###### Project name
`nix search` name: nixpkgs.pwsafe
<!--
The current version can be found easily with the same process than above for checking the master branch
If an open PR is present for the package, take this version as the current one and link to the PR
-->
current version: 3.53.0
desired version:

###### Notify maintainers
<!--
Search your package here: https://search.nixos.org/packages?channel=unstable
If no maintainer is listed for your package, tag the person that last updated the package
-->

maintainers: @c0bw3b @pjones 

This fails to build for me, crashing out on the build of `wxwidgets` (which is presumably where the link to pull 73145 comes from).  Given this is apparently a failure in a dependency, it may be that `pwsafe` will automatically start building again once this is resolved.  If that's so, then this report is redundant; in which case, apologies.  But I'll add it nonetheless in case the fix is not that automatic (I'm a bit vague about how fixes propagate around the wonderful wonderland of Nix).

    clang++ -mmacosx-version-min=10.7 -c -o htmldll_helpdata.o  -D__WXOSX_COCOA__      -DWXBUILDING      -I./src/regex  -DWXUSINGDLL -DWXMAKINGDLL_HTML -dynamic -fPIC -DPIC -Wall -Wundef -Wunused-parameter -Wno-ctor-dtor-privacy -Woverloaded-virtual -Wno-deprecated-declarations -D_FILE_OFFSET_BITS=64 -I/private/tmp/nix-build-wxwidgets-3.0.4.drv-0/source/lib/wx/include/osx_cocoa-unicode-3.0 -I./include -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=1 -O2 -fno-strict-aliasing -fno-common -fvisibility=hidden -fvisibility-inlines-hidden ./src/html/helpdata.cpp
    ./src/osx/cocoa/mediactrl.mm:470:9: fatal error: 'AVFoundation/AVFoundation.h' file not found
    #import <AVFoundation/AVFoundation.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    make: *** [Makefile:39594: mediadll_osx_cocoa_mediactrl.o] Error 1
    make: *** Waiting for unfinished jobs....
    builder for '/nix/store/wm81bmbkzk2ziyk5v3spdz4647n1v5vn-wxwidgets-3.0.4.drv' failed with exit code 2
    cannot build derivation '/nix/store/y5b7m6vlp8l5046sf7g41bz88w2dv9gn-pwsafe-3.53.0.drv': 1 dependencies couldn't be built
    error: build of '/nix/store/y5b7m6vlp8l5046sf7g41bz88w2dv9gn-pwsafe-3.53.0.drv' failed

This is with the unstable channel fully updated:

    $ nix-env -qa --drv-path --out-path pwsafe
    pwsafe-3.53.0  /nix/store/y5b7m6vlp8l5046sf7g41bz88w2dv9gn-pwsafe-3.53.0.drv  /nix/store/lvikq70mprd98ny90glpsh04llzs77nz-pwsafe-3.53.0

Platform: macOS 10.15.7 (Catalina).


###### Note for maintainers

Please tag this issue in your PR.

--- a/pkgs/applications/misc/pwsafe/default.nix
+++ b/pkgs/applications/misc/pwsafe/default.nix
@@ -19,9 +19,13 @@
 , curl
 , openssl
 , file
+, darwin
 , gitUpdater
 }:
 
+let
+  inherit (darwin.apple_sdk.frameworks) Cocoa;
+in
 stdenv.mkDerivation rec {
   pname = "pwsafe";
   version = "1.16.0"; # do NOT update to 3.x Windows releases
@@ -33,6 +37,8 @@ stdenv.mkDerivation rec {
     hash = "sha256-5/TOg+hiy22vlPJHheE638abhS3B5Jrul0Umgwu+gi0=";
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [
     cmake
     gettext
@@ -42,24 +48,29 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
+    wxGTK32
+    curl
+    qrencode
+    openssl
+    xercesc
+    file
+  ] ++ lib.optionals stdenv.isLinux [
     libXext
     libXi
     libXt
     libXtst
-    wxGTK32
-    curl
-    qrencode
     libuuid
-    openssl
-    xercesc
     libyubikey
     yubikey-personalization
-    file
+  ] ++ lib.optionals stdenv.isDarwin [
+    Cocoa
   ];
 
   cmakeFlags = [
     "-DNO_GTEST=ON"
     "-DCMAKE_CXX_FLAGS=-I${yubikey-personalization}/include/ykpers-1"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "-DNO_YUBI=ON"
   ];
 
   postPatch = ''
@@ -79,6 +90,8 @@ stdenv.mkDerivation rec {
     for f in $(grep -Rl /usr/bin/ .) ; do
       substituteInPlace $f --replace /usr/bin/ ""
     done
+  '' + lib.optionalString stdenv.isDarwin ''
+    substituteInPlace src/ui/cli/CMakeLists.txt --replace "uuid" ""
   '';
 
   installFlags = [ "PREFIX=${placeholder "out"}" ];
@@ -99,7 +112,7 @@ stdenv.mkDerivation rec {
     '';
     homepage = "https://pwsafe.org/";
     maintainers = with maintainers; [ c0bw3b pjones ];
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     license = licenses.artistic2;
   };
 }

